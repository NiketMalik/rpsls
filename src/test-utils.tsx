import { FC, ReactElement } from "react";
import { render, RenderOptions } from "@testing-library/react";
import { THEME } from "constants/theme";

import { StoreProvider } from "context/storeContext";
import { ThemeProvider } from "@emotion/react";

const WithGlobalProviders: FC = ({ children }) => {
  return (
    <StoreProvider>
      <ThemeProvider theme={THEME}>{children}</ThemeProvider>
    </StoreProvider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">,
) => render(ui, { wrapper: WithGlobalProviders, ...options });

export * from "@testing-library/react";
export { customRender as render };
