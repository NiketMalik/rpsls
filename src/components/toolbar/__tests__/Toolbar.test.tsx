import React from "react";
import { render, screen } from "test-utils";

import { Toolbar } from "../";

jest.mock("../components/rules", () => ({ Rules: () => null }));
jest.mock("../components/volume", () => ({ Volume: () => null }));

describe("Toolbar", () => {
  it("renders without error", () => {
    const { container } = render(<Toolbar />);
    const toolbarEle = screen.getByTestId("toolbar");
    expect(toolbarEle).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
