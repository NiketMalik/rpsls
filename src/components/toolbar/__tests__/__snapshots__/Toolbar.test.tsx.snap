// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar renders without error 1`] = `
<div>
  <div
    class="d-flex align-items-center justify-content-center justify-content-md-end css-4nq16i"
    data-testid="toolbar"
  />
</div>
`;
